File: 2022/day04/day04.py
import re

with open('input.txt', 'r') as f:
	lines = f.read().splitlines()

part_1 = part_2 = 0
for line in lines:
	ints = [*map(int, re.findall(r'\d+', line))]
	sets = set(range(ints[0], ints[1] + 1)), set(range(ints[2], ints[3] + 1))
	if sets[0].issubset(sets[1]) or sets[0].issuperset(sets[1]):
		part_1 += 1
	if sets[0] & sets[1]:
		part_2 += 1

print(f'Part 1: {part_1}')
print(f'Part 2: {part_2}')

File: 2022/day04/day04.rb
require 'set'

f = File.readlines("input.txt", chomp: true)

part_1 = 0
part_2 = 0

f.each do |line|
	nbs = line.scan(/\d+/).map(&:to_i)
	set_1, set_2 = Set.new(nbs[0]..nbs[1]), Set.new(nbs[2]..nbs[3])
	if set_1.subset?(set_2) || set_1.superset?(set_2)
		part_1 += 1
	end
	if set_1.intersect?(set_2)
		part_2 += 1
	end
end

puts "Part 1: #{part_1}"
puts "Part 2: #{part_2}"
